### PR TITLE
feat: enable verify certs

### PR DESCRIPTION
**Pull request by Daniel-Bloom-dfinity**
icx-proxy should support the certificate verification, same as [service worker](https://github.com/dfinity/ic/blob/master/typescript/service-worker/src/sw/http_request.ts#L257).

This is mostly just a port of that code.

### CODE REVIEW

**Comment by nomeata** [comment]
It's good to have the code, but what is the use case? The certificate can help against an attackers who can change the response, but with the code as it is now, just removing the header will make ic-proxy accept the request. Is there more coming (a strict mode where the header is mandatory maybe?)

File: src/main.rs [hunk 1]
```diff
@@ -336,6 +399,34 @@ async fn forward_request(
...
+                Err(e) => {
+                    return Ok(Response::builder()
+                        .status(StatusCode::INTERNAL_SERVER_ERROR)
+                        .body(format!("Certificate validation failed: {}", e).into())
+                        .unwrap());
+                }
+            },
+            (Some(_), _) | (_, Some(_)) => false,
+            // Canisters don't have to provide certified variables
+            (None, None) => true,
```
Comment (farazshaikh) [inline src/main.rs:422]: this is problematic as @nomeata has pointed out. The service worker code 
https://github.com/dfinity/ic/blob/master/typescript/service-worker/src/sw/http_request.ts#L287
treats the absence of IC-CERTIFICATE as invalid responses. Here the icx-proxy is treating it as the valid body. So there is a  semantic difference in this case.

// Canisters don't have to provide certified variables
Given this comment, I think we need a bit more discussion around this.

Comment (Daniel-Bloom-dfinity) [reply src/main.rs:422]: The long term plan is to close this gap, but in order to not break existing canisters, this fallback has to remain in place for now.

**Comment by Daniel-Bloom-dfinity** [comment]
> Is there more coming (a strict mode where the header is mandatory maybe?)

Essentially yes, but the design is still being finalized.

**Comment by paulyoung** [comment]
Or perhaps a more pertinent question is; does `icx-proxy` still verify certs for `.raw.ic0.app`?

**Pull request merged**